src/_pytest/config/__init__.py

```diff
@@ -429,7 +429,13 @@ def _getconftestmodules(self, path):
         # and allow users to opt into looking into the rootdir parent
         # directories instead of requiring to specify confcutdir
         clist = []
-        for parent in directory.realpath().parts():
+        
+        p = directory
+        if sys.platform != 'win32':
```

Comment:
Please move this logic to a function in `_pytest.pathlib` and reuse it in the other places, including adding the issue number in the docstring. 👍 

```python
def real_path(p: Union[str, py.path.local, Path]):
    """Returns the full path name, but not following symbolic links on Windows (see #5965)."""
    p = py.path.local(p)
    if sys.platform != 'win32':
        p = p.realpath()
    return p
```